File: Daily Challenge/C/Floating Point Place Values 24.7.2025.c
#include <stdio.h>
#include <string.h>
#include <math.h>

int main() {
    char s[100];
    scanf("%s", s);
    char *dot = strchr(s, '.');
    int len = strlen(s), decLen = dot - s, fracLen = len - decLen - 1;

    int a = decLen - 1, b = 1;

    for (int i = 0; i < decLen; i++) {
        if (s[i] == '0') {
            printf("0 ");
        } else {
            printf("%d ", (s[i] - '0') * (int)pow(10, a));
        }
        a--;
    }

    for (int i = decLen + 1; i < len; i++) {
        if (s[i] == '0') {
            printf("0 ");
        } else {
            printf("%d/%d ", s[i] - '0', (int)pow(10, b));
        }
        b++;
    }
    return 0;
}

File: Daily Challenge/Java/Floating Point Place Values 24.7.2025.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        int dot = s.indexOf(".");
        int a = dot - 1, b = 1;

        for (int i = 0; i < dot; i++) {
            if (s.charAt(i) == '0') {
                System.out.print("0 ");
            } else {
                System.out.print((s.charAt(i) - '0') * (int)Math.pow(10, a) + " ");
            }
            a--;
        }

        for (int i = dot + 1; i < s.length(); i++) {
            if (s.charAt(i) == '0') {
                System.out.print("0 ");
            } else {
                System.out.print((s.charAt(i) - '0') + "/" + (int)Math.pow(10, b) + " ");
            }
            b++;
        }
    }
}